src/vs/workbench/api/browser/mainThreadEditorTabs.ts

```diff
@@ -27,31 +31,133 @@ export class MainThreadEditorTabs {
 
 		this._proxy = extHostContext.getProxy(ExtHostContext.ExtHostEditorTabs);
 
-		this._dispoables.add(editorService.onDidEditorsChange(this._pushEditorTabs, this));
-		this._editorGroupsService.whenReady.then(() => this._pushEditorTabs());
+		// Queue all events that arrive on the same event loop and then send them as a batch
+		this._dispoables.add(editorService.onDidEditorsChange((e) => {
+			this._queuedEvents.push(e);
+			if (this._queuedEvents.length === 1) {
+				queueMicrotask(() => {
```

Comment:
Is it worth making this reusable maybe? We already have a similar thing called `RunOnceWorker` which is based on the `RunOneScheduler`, however `setTimeout` based. This could be a variant (which unfortunately is not cancelable though), like `RunMicrotaskWorker`?

Reply:
I'm not too sure the use case of these workers so I'm hesitant to add code just to have it. For now I've added `MicroTaskEmitter` for events which is my current usecase.

Reply:
Yeah I much prefer to do the microtask queueing where the event is emitted than where it is received.